streams/dask.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import absolute_import, division, print_function
+
+from operator import getitem
+
+from tornado import gen
+
+from distributed.client import default_client
+
+from . import core
+
+
+class DaskStream(core.Stream):
+    def map(self, func, *args, **kwargs):
+        """ Apply a function to every element in the stream """
+        return map(func, self, args=args, **kwargs)
+
+    def gather(self):
+        return gather(self)
+
+    def accumulate(self, func, start=core.no_default, returns_state=False):
+        """ Accumulate results with previous state """
+        return scan(func, self, start=start, returns_state=returns_state)
+
+    scan = accumulate
+
+    def zip(self, *other):
+        """ Combine two streams together into a stream of tuples """
+        return zip(self, *other)
+
+
+class map(DaskStream):
+    def __init__(self, func, child, args=(), **kwargs):
+        self.func = func
+        self.kwargs = kwargs
+        self.args = args
+
+        DaskStream.__init__(self, child)
+
+    def update(self, x, who=None):
+        client = default_client()
+        result = client.submit(self.func, x, *self.args, **self.kwargs)
+        return self.emit(result)
+
+
+class scan(DaskStream):
+    def __init__(self, func, child, start=core.no_default, returns_state=False):
+        self.func = func
+        self.state = start
+        self.returns_state = returns_state
+        DaskStream.__init__(self, child)
+
+    def update(self, x, who=None):
+        if self.state is core.no_default:
+            self.state = x
+            return self.emit(self.state)
+        else:
+            client = default_client()
+            result = client.submit(self.func, self.state, x)
```

Comment:
The scheduler doesn't contain parallel algorithms like this, it just executes what we told it to do.  In a case of `scan(add)` we're giving it a sequential algorithm, which it will execute faithfully on parallel hardware.  Most of the benefit of parallelism will probably come from maps.